test/distributed/org/apache/cassandra/distributed/test/InternodeEncryptionEnforcementTest.java

```diff
@@ -40,6 +50,104 @@
 
 public final class InternodeEncryptionEnforcementTest extends TestBaseImpl
 {
+
+    @Test
+    public void testInboundConnectionsAreRejectedWhenAuthFails() throws IOException, InterruptedException
+    {
+        Cluster.Builder builder = createCluster(RejectInboundConnections.class);
+
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        try (Cluster cluster = builder.start())
+        {
+            executorService.submit(() -> openConnections(cluster));
+
+            /*
+             * instance (1) should not connect to instance (2) as authentication fails;
+             * instance (2) should not connect to instance (1) as authentication fails.
+             */
+            IIsolatedExecutor.SerializableCallable<Boolean> callable = () ->
+            {
+                InboundMessageHandlers inbound = getOnlyElement(MessagingService.instance().messageHandlers.values());
+                assertEquals(0, inbound.count());
+                OutboundConnections outbound = getOnlyElement(MessagingService.instance().channelManagers.values());
+                assertTrue(!outbound.small.isConnected() && !outbound.large.isConnected() && !outbound.urgent.isConnected());
+
+                // Verify that the failure is due to authentication failure
+                final RejectInboundConnections authenticator = (RejectInboundConnections) DatabaseDescriptor.getInternodeAuthenticator();
+                assertTrue(authenticator.authenticationFailed);
+                return true;
+            };
+
+            // Wait for cluster to get started
+            Thread.sleep(3000);
+            assertTrue(cluster.get(1).callsOnInstance(callable).call());
```

Comment:
The `callable` always returns `true`. Why are the assertions needed? If the assertions are not needed, what about just `runOnInstance`?